screens/user-list.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { useState } from 'react';
+import { View, Text, FlatList, StyleSheet, ActivityIndicator } from 'react-native';
+import { useQuery } from '@apollo/client';
+import { getUsers } from '../components/utils/graphql-requests';
+
+export const UserListScreen = () => {
+  const [offset, setOffset] = useState(0);
+
+  const [users, setUser] = useState([]);
+
+  const onCompleted = (data: any) => {
+    console.log(data);
+    setUser((prev) => [...prev, ...data?.users.nodes] as any);
+  };
+  const { loading, error, data } = useQuery(getUsers, {
+    variables: { offset: offset },
+    notifyOnNetworkStatusChange: true,
+    onCompleted: onCompleted,
+  });
+
+  const handleEndReached = () => {
+    if (users.length + 10 < data?.users.count) {
+      setOffset(offset + 10);
+    }
+  };
+
+  const renderItem = (item: any) => {
+    return (
+      <>
+        <Text style={styles.item}>{`Nome : ${item.item.name}`}</Text>
+        <Text style={styles.item}>{`Email : ${item.item.email}`}</Text>
+      </>
+    );
+  };
+
+  return (
+    <View style={styles.container}>
+      <FlatList
+        data={users}
+        onEndReached={handleEndReached}
+        keyExtractor={(_item, index) => String(index)}
```

Comment:
Não costuma ser uma boa usar index como key de atualização. O React mesmo contra-recomenda, veriicar [aqui](https://reactjs.org/docs/lists-and-keys.html) para mais detalhes, mas basicamente um identificador único pra cada elemento e a melhor opção. No caso, o id de usuário.